### PR TITLE
Add pod annotations backup lightning chart

### PR DESCRIPTION
**Pull request by jlerche**
### What problem does this PR solve? 
tidb-backup and tidb-lightning charts do not allow for extra pod annotations. Pod annotations are necessary to be compatible with kube2iam

tidb-lightning chart assumes importer backend

### What is changed and how does it work?
* Pod annotations pass-through added to tidb-backup and tidb-lightning
* Option to specify tidb-lightning backend `importer | tidb`
* small fixes in usability

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note

 ```

### CODE REVIEW

File: charts/tidb-lightning/templates/job.yaml [hunk 1]
```diff
@@ -37,7 +40,7 @@ spec:
           mountPath: /etc/rclone
         - name: data
           mountPath: /data
-      {{- if and (ne .Values.timezone "UTC") (ne .Values.timezone "") }}
+      {{ if and (ne .Values.timezone "UTC") (ne .Values.timezone "") }}
```
Comment (gregwebs) [inline charts/tidb-lightning/templates/job.yaml]: why no dash?

Comment (jlerche) [reply charts/tidb-lightning/templates/job.yaml]: I was seeing the following:
```volumeMounts:
        - name: credentials
          mountPath: /etc/rclone
        - name: data
          mountPath: /datarestartPolicy: Never      # if lightning fails, manual intervention is required so no restart```

Comment (tennix) [reply charts/tidb-lightning/templates/job.yaml]: Does the latest tidb-lightning exit with 0 when loading successfully now?

File: charts/tidb-lightning/values.yaml [hunk 2]
```diff
@@ -47,9 +47,13 @@ resources: {}
 
 nodeSelector: {}
 
+annotations: {}
+
 tolerations: []
 affinity: {}
 
+backend: importer # importer | tidb
```
Comment (tennix) [inline charts/tidb-lightning/values.yaml]: Which tidb version supports specifying the backend?

File: charts/tidb-lightning/templates/job.yaml (hunk 1 above)
Comment (jlerche) [reply charts/tidb-lightning/templates/job.yaml]: @tennix that is what I saw, yes.

Comment (tennix) [reply charts/tidb-lightning/templates/job.yaml]: Then I think we can remove the fail-fast hack script.

File: charts/tidb-lightning/values.yaml (hunk 2 above)
Comment (jlerche) [reply charts/tidb-lightning/values.yaml]: It was tested with 3.0.3.

File: charts/tidb-lightning/templates/job.yaml (hunk 1 above)
Comment (jlerche) [reply charts/tidb-lightning/templates/job.yaml]: @tennix I removed the script as well as the fail fast option

File: charts/tidb-lightning/templates/scripts/_start_lightning.sh.tpl [hunk 3]
```diff
@@ -16,18 +16,13 @@ fi
...
+{{- else if eq .Values.backend "tidb" }}
+    --backend tidb \
+{{- end }}
     --server-mode=false \
     --tidb-user={{ .Values.targetTidbCluster.user | default "root" }} \
     --tidb-host={{ .Values.targetTidbCluster.name }}-tidb.{{ .Values.targetTidbCluster.namespace | default .Release.Namespace }} \
     --d=${data_dir} \
     --config=/etc/tidb-lightning/tidb-lightning.toml
-
-if [ $? != 0 ]; then
```
Comment (gregwebs) [inline charts/tidb-lightning/templates/scripts/_start_lightning.sh.tpl]: If we aren't checking the exit code then we need `set -e`. Best to have `set -euo pipefail`

**Comment by gregwebs** [comment]
I don't think we have any e2e tests for this. This was tested manually by Jacob. merging now.

**Pull request merged**